hq_superset/views.py

```diff
@@ -141,17 +152,47 @@ def array_is_falsy(array_values):
     return array_values
 
 
-def refresh_hq_datasource(domain, datasource_id, display_name):
+def trigger_datasource_refresh(domain, datasource_id, display_name):
+    if AsyncImportHelper(domain, datasource_id).is_import_in_progress():
+        flash("The datasource is already being imported in the background. "
+              "Please wait for it to finish before retrying",
+              "warning")
+        return redirect("/tablemodelview/list/")
+
+    provider = superset.appbuilder.sm.oauth_remotes["commcare"]
+    token = get_valid_cchq_oauth_token()
+    path, size = download_datasource(provider, token, domain, datasource_id)
+    datasource_defn = get_datasource_defn(provider, token, domain, datasource_id)
+    if size < ASYNC_DATASOURCE_IMPORT_LIMIT_IN_BYTES:
+        response = refresh_hq_datasource(
+            domain, datasource_id, display_name, path, datasource_defn, None
+            )
+        os.remove(path)
+        return response
+    else:
+        limit_in_mb = int(ASYNC_DATASOURCE_IMPORT_LIMIT_IN_BYTES / 1000000)
```

Comment:
Not worth the hassle of changing, but useful to know: Python 3.6+ supports [using "_" as a thousands separator](https://peps.python.org/pep-0515/) to make large numbers more readable, like
```python
>>> foo = 10_000_000
>>> foo
10000000
```
(Also works for hex (`0xCAFE_F00D`) and binary (`0b_0011_1111_0100_1110`).)

Reply:
That's neat!